### PR TITLE
codeintel: extra trace/honeycomb data in precise-code-intel-worker

### PR DESCRIPTION
**Pull request by Strum355**
To be merged after https://github.com/sourcegraph/sourcegraph/pull/27964

### CODE REVIEW

File: enterprise/cmd/precise-code-intel-worker/internal/worker/handler_test.go [hunk 1]
```diff
@@ -254,7 +255,7 @@ func TestHandleCloneInProgress(t *testing.T) {
 		gitserverClient: gitserverClient,
 	}
 
-	requeued, err := handler.handle(context.Background(), upload)
+	requeued, err := handler.handle(context.Background(), upload, func(fields ...log.Field) {})
```
Comment (efritz) [inline enterprise/cmd/precise-code-intel-worker/internal/worker/handler_test.go:258]: This is also a `observable.TraceLogger`

Comment (efritz) [reply enterprise/cmd/precise-code-intel-worker/internal/worker/handler_test.go:258]: or whatever I call my packages 🙄

File: enterprise/cmd/precise-code-intel-worker/internal/worker/iface.go [hunk 2]
```diff
@@ -52,14 +52,14 @@ type LSIFStore interface {
...
 	WriteMeta(ctx context.Context, bundleID int, meta precise.MetaData) error
-	WriteDocuments(ctx context.Context, bundleID int, documents chan precise.KeyedDocumentData) error
-	WriteResultChunks(ctx context.Context, bundleID int, resultChunks chan precise.IndexedResultChunkData) error
-	WriteDefinitions(ctx context.Context, bundleID int, monikerLocations chan precise.MonikerLocations) error
-	WriteReferences(ctx context.Context, bundleID int, monikerLocations chan precise.MonikerLocations) error
-	WriteImplementations(ctx context.Context, bundleID int, monikerLocations chan precise.MonikerLocations) error
-	WriteDocumentationPages(ctx context.Context, upload dbstore.Upload, repo *types.Repo, isDefaultBranch bool, documentation chan *precise.DocumentationPageData, repositoryNameID int, languageNameID int) error
-	WriteDocumentationPathInfo(ctx context.Context, bundleID int, documentation chan *precise.DocumentationPathInfoData) error
-	WriteDocumentationMappings(ctx context.Context, bundleID int, mappings chan precise.DocumentationMapping) error
+	WriteDocuments(ctx context.Context, bundleID int, documents chan precise.KeyedDocumentData) (count int64, err error)
```
Comment (efritz) [inline enterprise/cmd/precise-code-intel-worker/internal/worker/iface.go:55]: Could these be `uints` instead? Should never be negative unless something really bad happens and we lose result chunks (but can accurately count them) when we call this method.

**Review by efritz — APPROVED** [review]

File: enterprise/cmd/precise-code-intel-worker/internal/worker/handler_test.go (hunk 1 above)
Comment (Strum355) [reply enterprise/cmd/precise-code-intel-worker/internal/worker/handler_test.go:258]: Not sure how this is meant to be used :thonk: afaik this is the shortest way to create a non nil one

Comment (efritz) [reply enterprise/cmd/precise-code-intel-worker/internal/worker/handler_test.go:258]: Oh sorry I read this as a type at a glance 🤦

**Pull request merged**